projects/views.py

```diff
@@ -39,4 +41,68 @@ def get(self, request, id):
             return JsonResponse({'result': result}, status=200)
 
         except Project.DoesNotExist:
-            return JsonResponse({'messages': 'DOES_NOT_EXIST'}, status=404)
\ No newline at end of file
+            return JsonResponse({'messages': 'DOES_NOT_EXIST'}, status=404)
+
+class ProjectListView(View):
+    def get(self, request):
+        queries = request.GET
+        progress_min  = queries.get('progressMin')
+        progress_max  = queries.get('progressMax')
+        amount_min    = queries.get('amountMin')
+        amount_max    = queries.get('amountMax')
+        category      = queries.get('category')
+        status        = queries.get('status')
+        sort_criteria = queries.get('sorted', 'default')
+        search        = queries.get('search')
+
+        filter_set = {
+            'progress__gte'      : progress_min,
+            'progress__lte'      : progress_max,
+            'funding_amount__gte': amount_min,
+            'funding_amount__lte': amount_max,
+            'category__name'     : category,
+            'status'             : status
+        }
+
+        filter_set = { k: v for k, v in filter_set.items() if v }
+
+        sortby_set = {
+            'default': '-created_at',
+            'latest' : '-created_at',
+            'people' : '-funding_count',
+            'amount' : '-funding_amount',
+            'old'    : 'end_date'
+        }
+
+        project_list = Project.objects.all().order_by('-id')\
+                                            .select_related('category', 'creater')\
+                                            .prefetch_related('donation_set')\
+                                            .annotate(funding_amount=Coalesce(Sum('donation__funding_option__amount'), Decimal(0)))\
+                                            .annotate(funding_count=Coalesce(Count('donation'), 0))\
+                                            .annotate(progress=100 * F('funding_amount')/F('target_fund'))\
+                                            .annotate(status=Case(When(end_date__lt=datetime.now(), then=Value("done")),
+                                                                  When(launch_date__gt=datetime.now(), then=Value("scheduled")),
+                                                                  default=Value("ing")))
+
+        project_list = project_list.filter(**filter_set).order_by(sortby_set[sort_criteria])
```

Comment:
filter와 order_by을 나누어서 진행하는데, 특정 조건으로 안 나뉘는 것 같은데 그러면 둘을 합칠 수 있지 않을까요?
```suggestion
        project_list = Project.objects.filter(**filter_set).order_by(sortby_set[sort_criteria])\
                                            .select_related('category', 'creater')\
                                            .prefetch_related('donation_set')\
                                            .annotate(funding_amount=Coalesce(Sum('donation__funding_option__amount'), Decimal(0)))\
                                            .annotate(funding_count=Coalesce(Count('donation'), 0))\
                                            .annotate(progress=100 * F('funding_amount')/F('target_fund'))\
                                            .annotate(status=Case(When(end_date__lt=datetime.now(), then=Value("done")),
                                                                  When(launch_date__gt=datetime.now(), then=Value("scheduled")),
                                                                  default=Value("ing")))

```

Reply:
오호! 감사합니다 적용해서 업데이트하겠습니다!